dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ConfigCenterConfig.java

```diff
@@ -183,13 +213,16 @@ public void setAppConfigFile(String appConfigFile) {
         this.appConfigFile = appConfigFile;
     }
 
-    @Parameter(key = Constants.CONFIG_APPNAME_KEY, useKeyAsProperty = false)
-    public String getAppName() {
-        return appName;
+    public ApplicationConfig getApplication() {
+        if (application != null && !application.isValid()) {
+            throw new IllegalStateException("Name is required for ApplicationConfig, for example, in XML it looks like <dubbo:application name=\"...\" />.");
+        }
+        return application;
     }
 
-    public void setAppName(String appName) {
-        this.appName = appName;
+    public void setApplication(ApplicationConfig application) {
+        ConfigManager.getInstance().setApplication(application);
```

Comment:
should check whether given **application** is valid or not like we are doing in getApplication method?